### PR TITLE
Feature: Add Solar Endpoints and Objects

### PR DESCRIPTION
**Pull request by alecdoconnor**
Add Tesla Solar Endpoints, objects, and update second view controller to view this data.
Also add Combine, PromiseKit, and RX implementation

### CODE REVIEW

File: Sources/Extensions/Combine/TeslaSwift+Combine.swift [hunk 1]
```diff
@@ -193,6 +193,83 @@ extension TeslaSwift {
...
+        
+        return future
+    }
+
+    public func getEnergySiteInfo(siteID: String) -> Future<EnergySiteInfo, Error> {
+        
+        let future = Future<EnergySiteInfo, Error> { (subscriber: @escaping (Result<EnergySiteInfo, Error>) -> Void) in
+            
+            self.getEnergySiteInfo(siteID: siteID, completion: self.resultify(subscriber: subscriber))
+
```
[the remark this answers is not in the document]
Comment (alecdoconnor) [reply Sources/Extensions/Combine/TeslaSwift+Combine.swift:224]: Following pattern of rest of file

File: Sources/Extensions/Combine/TeslaSwift+Combine.swift [hunk 2]
```diff
@@ -193,6 +193,83 @@ extension TeslaSwift {
...
+            
+        }
+        
+        return future
+    }
+
+    public func getEnergySiteInfo(siteID: String) -> Future<EnergySiteInfo, Error> {
+        
+        let future = Future<EnergySiteInfo, Error> { (subscriber: @escaping (Result<EnergySiteInfo, Error>) -> Void) in
+
```
[the remark this answers is not in the document]
Comment (alecdoconnor) [reply Sources/Extensions/Combine/TeslaSwift+Combine.swift:222]: Following pattern of rest of file

**Comment by jonasman** [comment]
Thanks for the PR, i will need to test a few things before merging it. but so far looks good

**Comment by jonasman** [comment]
Sorry, can you fix the conflicts so i can have a look?

**Pull request merged**